pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -640,7 +640,8 @@ func (s *DiscoveryServer) loadAssignmentsForClusterIsolated(proxy *model.Proxy,
 	// will update "XdsConnection.Clusters", we might accidentally send EDS updates for STRICT_DNS cluster. This check gaurds
 	// against such behavior and returns nil. When the updated cluster warms up in Envoy, it would update with new endpoints
 	// automatically.
-	if svc.Resolution != model.ClientSideLB {
+	// Gateways use EDS for Passthrough cluster. So we should allow Passthrough here.
+	if svc.Resolution == model.DNSLB {
```

Comment:
should we check for gateway here? otherwise sidecar passthrough can break here?

Reply:
It should not break. Only DNS will break

Reply:
But if you think it is better to check for gateways here - I am open to it. LMK

Reply:
oh right, makes sense